src/Backend/One/OneToolchain.ts

```diff
@@ -14,33 +14,188 @@
  */
 
+import {execSync} from 'child_process';
+import * as vscode from 'vscode';
+
 import {Backend} from '../Backend';
 import {Command} from '../Command';
 import {Compiler} from '../Compiler';
 import {Executor} from '../Executor';
-import {Toolchains} from '../Toolchain';
+import {PackageInfo, ToolchainInfo, Toolchains} from '../Toolchain';
+import {DebianArch, DebianRepo, DebianToolchain} from '../ToolchainImpl/DebianToolchain';
+import {Version} from '../Version';
+
+class OneDebianToolchain extends DebianToolchain {
+  run(cfg: string): Command {
+    this.prepare();
+    let cmd = new Command('onecc-docker');
+    cmd.push('-C');
+    cmd.push(cfg);
+    return cmd;
+  }
+}
 
 class OneCompiler implements Compiler {
   private readonly toolchainTypes: string[];
+  private readonly toolchainName: string;
+  private readonly debianRepo: DebianRepo;
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker';
+    this.toolchainTypes = ['latest'];
+
+    this.debianRepo = new DebianRepo(
+        'http://ppa.launchpad.net/one-compiler/onecc-docker/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
+  parseVersion(version: string): Version {
+    if (!version.trim()) {
+      throw Error('Invalid version format.');
+    }
+
+    let _version = version;
+    let option = '';
+
+    const optionIndex = version.search(/[~+-]/);
+    if (optionIndex !== -1) {
+      option = version.slice(optionIndex);
+      _version = version.slice(0, optionIndex);
+    }
+
+    const splitedVersion = _version.split('.');
+
+    if (splitedVersion.length > 3) {
+      throw Error('Invalid version format.');
+    }
+
+    let major: number|string;
+    let minor: number|string;
+    let patch: number|string;
+
+    [major = '0', minor = '0', patch = '0'] = _version.split('.');
+
+    const epochIndex = major.search(/:/);
+    if (epochIndex !== -1) {
+      major = major.slice(epochIndex + 1);
+    }
+
+    major = Number(major);
+    minor = Number(minor);
+    patch = Number(patch);
+
+    if (isNaN(major) || isNaN(minor) || isNaN(patch)) {
+      throw Error('Invalid version format.');
+    }
+    return new Version(major, minor, patch, option);
+  }
+
   getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+    if (_toolchainType !== 'latest') {
+      throw Error(`Invalid toolchain type : ${_toolchainType}`);
+    }
+
+    if (_start < 0) {
+      throw Error(`wrong start number: ${_start}`);
+    }
+    if (_count < 0) {
+      throw Error(`wrong count number: ${_count}`);
+    }
+    if (_count === 0) {
+      return [];
+    }
+
+    try {
+      execSync(`apt-cache show ${this.toolchainName}`);
+    } catch (error) {
+      throw Error(`Getting ${this.toolchainName} package list is failed`);
```

Comment:
Could you not throw an error? If so, the extension will quit.

Reply:
No, if it throw an error, prerequisites is executed.

Reply:
> No, if it throw an error, prerequisites is executed.

Where is the line?

Reply:
Aha... I understand how it is. IMO, returning something to a caller would be better than throwing, but it's not about this PR.